### PR TITLE
Fix the ground slab horizontal insulation thickness issue (Issues 7881 and 8800)

### LINKED ISSUE #7881: Site:GroundDomain:Slab is not using the material thickness of the horizontal insulation material
Issue overview
--------------
User notes:
1. If I add horizontal insulation to an in-grade slab using the Site:GroundDomain:Slab inputs then changing its thickness appears to have no effect on the results. 25mm of insulation results in exactly the same heating energy as 250mm of insulation. Adding insulation, or changing the material completely to one with different conductivity does change the results, so it is picking up on the material - just not its thickness. Unless I'm missing something, I'm fairly sure this is not how it's supposed to work?

2. Also, on a side note Figure 1.8: On-grade configuration in the InputOutputReference for v9.2 is showing the diagram for the Site:GroundDomain:Basement model rather than the on-grade slab.

Developer response:
Searching the source code shows that the horizontal insulation material thickness is only used for the basement domain. It appears on quick inspection that the slab horizontal insulation may be incorrectly using the thickness of the vertical insulation material. The other properties of the horizontal insulation material are flowing into the correct place.

### Details
Some additional details for this issue (if relevant):
 - Version of EnergyPlus 9.3.0-e004ffb2c9
 - Helpdesk ticket number 15426

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Defect file added EnergyPlusDevSupport\DefectFiles
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### LINKED ISSUE #8800: Vertical insulation thickness misused as Horizontal Insulation thickness in ground coupled slabs domain partitioning
Issue overview
--------------
There are a few reports regarding the Site:GroundDomain:Slab object’s horizontal insulation thickness not affecting the energy results (e.g. heating energy) whereas it is actually expected to do so. This problem is observed in both InGrade slabs (see helpdesk ticket 15426, reported in Issue #7881, and helpdesk ticket 15723), and also OnGrade slabs (see helpdesk ticket 15733). 

An examination of the source code reveals that the input processing is done correctly. However, there is a code error in the domain partitioning/meshing that causes the vertical insulation thickness is misused for the horizontal insulation thickness. 

The issue would need a code change to get fixed. An initial code change was made and tested as a draft solution--after the change the horizontal insulation thickness is shown to be active and the associated energy calculation results would change accordingly. 

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version): All
 - Version of EnergyPlus (if using an intermediate build, include SHA): v9.3 - v9.5
 - Helpdesk tickets: 15426, 15723, 15733

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Defect file added to EnergyPlusDevSupport\DefectFiles
 - [x] Ticket added to Pivotal for defect (development team task)
 - [x] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by jcyuan2020**
Pull request overview
---------------------
 - Fixes #7881 and fixes #8800;
 - The original issue about an insensitive horizontal insulation thickness (in Site:GroundDomain:Slab) was reported in a couple of github issues and several helpdesk tickets. The users specified changed the insulation materials' thickness for the horizontal slab insulation but the resulted energy results (such as heating energy in winter, or cooling energy in summer) do not change even very drastic changes occurred between the different cases. The root cause was found to be a code error in the Finite Difference meshing for the slab calculation, where a horizontal thickness should have been used but instead in the original code the vertical insulation thickness was used. These would cause an invariant grid partitioning for the cells used for finite difference soil heat transfer calculation around the slab. The error is now corrected in this fix. 
 - Two unit tests with different horizontal thickness values were added to test out the grid partitioning portion of the related code;
 - An example file was created based upon an existing Site:GroundDomain:Slab test example IDF file, for an increased horizontal slab insulation thickness. 

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [x] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: src/EnergyPlus/PlantPipingSystemsManager.cc [hunk 1]
```diff
@@ -2966,7 +2966,7 @@ namespace PlantPipingSystemsManager {
 
             // Create Y-direction partitions
 
-            CellWidth = this->VertInsThickness;
+            CellWidth = this->HorizInsThickness;
```
Comment (mjwitte) [inline src/EnergyPlus/PlantPipingSystemsManager.cc:2971]: @jcyuan2020 I was puzzled that the regression tests showed no diffs, but then I see that the test files which have horizontal insulation, also have vertical insulation using the same material (so the insulation thickness is the same for both). If @Myoldmopar agrees with this change, then go ahead an prepare a unit test.

Comment (jcyuan2020) [reply src/EnergyPlus/PlantPipingSystemsManager.cc:2971]: @mjwitte Thanks! The defect files would demonstrate the difference, where the different horizontal insulation thickness is used.

Comment (Myoldmopar) [reply src/EnergyPlus/PlantPipingSystemsManager.cc:2971]: I agree this change looks reasonable and would suggest modifying an example file to use a different horizontal insulation thickness as well as forging ahead on a unit test.

Comment (jcyuan2020) [reply src/EnergyPlus/PlantPipingSystemsManager.cc:2971]: Thanks! :thumbsup:

File: testfiles/CMakeLists.txt [hunk 2]
```diff
@@ -44,6 +44,7 @@ add_simulation_test(IDF_FILE 5ZoneAirCooledDemandLimiting.idf EPW_FILE USA_IL_Ch
 add_simulation_test(IDF_FILE 5ZoneAirCooledDemandLimiting_FixedRateVentilation.idf EPW_FILE USA_IL_Chicago-OHare.Intl.AP.725300_TMY3.epw)
 add_simulation_test(IDF_FILE 5ZoneAirCooledDemandLimiting_ReductionRatioVentilation.idf EPW_FILE USA_IL_Chicago-OHare.Intl.AP.725300_TMY3.epw)
 add_simulation_test(IDF_FILE 5ZoneAirCooledWithCoupledInGradeSlab.idf EPW_FILE USA_IL_Chicago-OHare.Intl.AP.725300_TMY3.epw)
+add_simulation_test(IDF_FILE 5ZoneAirCooledWithCoupledInGradeSlab_HorizInsThickness.idf EPW_FILE USA_IL_Chicago-OHare.Intl.AP.725300_TMY3.epw)
```
Comment (Myoldmopar) [inline testfiles/CMakeLists.txt:47]: I actually would've preferred modifying an existing example file, but I'm not holding this up for now.  This is a great fix and the unit test looks good.  Building and testing locally now.

**Review by Myoldmopar — APPROVED** [review]

**Comment by Myoldmopar** [comment]
Builds and tests fine locally.  Good fix here @jcyuan2020, thanks.

**Pull request merged**